spec/controllers/sessions_controller_spec.rb

```diff
@@ -1,32 +1,59 @@
-# frozen_string_literal: true
-
 # sessions_controller_spec.rb
 require 'spec_helper'
 require 'rails_helper'
 RSpec.describe SessionsController do
-  describe '#create' do
-    it 'should successfully create a session' do
-      request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:google]
-
+ 
+    auth = {
+      'provider' => 'google',
+      'uid' => '12345',
+      'info' => {
+        'name' => 'John Doe',
+        'email' => 'johndoe@doe.com',
+        'location' => 'Doe World',
+        'image' => 'image_url'
+      },
+      'extra' => { 'raw_info' => { 'hd' => '@tamu.edu' } }
+    }
+  before do
+    request.env['omniauth.auth'] = auth
+  end
+ 
+  describe "#create" do
+ 
+    it "should successfully create a user" do
+      expect {
+        post :create
+      }.to change{ Authuser.count }.by(1)
+    end
+ 
+    it "should successfully create a session" do
       session[:user_id].should be_nil
+      post :create
+      session[:user_id].should_not be_nil
     end
-
-    it 'should redirect the user to the search page' do
-      response.should be_successful
+ 
+    it "should redirect the user to the root url" do
+      post :create
+      response.should redirect_to search_path
```

Comment:
Can you add an assertion to ensure that `session[:user_id]` is populated here? @deshantar99 this might be a good example for you to follow re: authentication before making requests.